GrablTracing.java

```diff
@@ -0,0 +1,106 @@
+package grakn.grabl_tracing;
+
+import grakn.grabl_tracing.protocol.TracingProto;
+import grakn.grabl_tracing.protocol.TracingServiceGrpc;
+import grakn.grabl_tracing.protocol.TracingServiceGrpc.TracingServiceBlockingStub;
+import grakn.grabl_tracing.protocol.TracingServiceGrpc.TracingServiceStub;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class GrablTracing implements AutoCloseable {
+    private final ManagedChannel channel;
+    private final TracingServiceBlockingStub tracingServiceBlockingStub;
+    private final TracingServiceStub tracingServiceStub;
+
+    public GrablTracing(String grablUri, String apiKey) {
+        //TODO authenticate with apiKey
+        channel = ManagedChannelBuilder.forTarget(grablUri).usePlaintext().build();
+        tracingServiceBlockingStub = TracingServiceGrpc.newBlockingStub(channel);
+        tracingServiceStub = TracingServiceGrpc.newStub(channel);
+    }
+
+    public Analysis analysis(String owner, String repo, String commit) {
+        return new Analysis(owner, repo, commit);
+    }
+
+    @Override
+    public void close() throws Exception {
+        channel.shutdown();
+        channel.awaitTermination(10, TimeUnit.SECONDS);
+        if (!channel.isTerminated()) {
+            channel.shutdownNow();
+        }
+    }
+
+    public class Analysis implements AutoCloseable {
+        private TracingSession session;
+
+        private Analysis(String owner, String repo, String commit) {
+            TracingProto.Analysis.Req req = TracingProto.Analysis.Req.newBuilder()
+                    .setOwner(owner)
+                    .setRepo(repo)
+                    .setCommit(commit)
+                    .build();
+            TracingProto.Analysis.Res res = tracingServiceBlockingStub.create(req);
+            String analysisId = res.getAnalysisId();
+            session = new TracingSession(tracingServiceStub, analysisId);
+        }
+
+        public Trace trace(String name, String tracker, int iteration) {
+            return new TraceImpl(name, tracker, iteration);
+        }
+
+        @Override
+        public void close() throws Exception {
+            session.close();
+        }
+
+        private class TraceImpl implements Trace {
+            private final UUID id;
```

Comment:
UUID sounds good to me. Is it what we decided to go with in based on our latest discussion? @adammitchelldev @haikalpribadi

Reply:
Yes, for now at least I've agreed it with Haikal. The alternative which we may choose to switch to before completion is to create a mixed key from the analysis id and some threaded ID allocation but for now UUID keeps it fairly simple.